lib/pure/strmisc.nim

```diff
@@ -85,6 +86,124 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+since (1, 5):
+  type ParseFloatOptions* = enum  ## Options for `parseFloatThousandSep`.
+    pfLeadingDot,    ## Allow leading dot, like ".9" and similar.
+    pfTrailingDot,   ## Allow trailing dot, like "9." and similar.
+    pfSepAnywhere,   ## Allow separator anywhere in between, like "9,9", "9,99".
+    pfDotOptional,   ## Allow "9", "-0", integers literals, etc.
+    pfEmptyString    ## Allow "" to return 0.0, so you do not need to do
+                     ## try: parseFloatThousandSep(str) except: 0.0
+
+  func parseFloatThousandSep*(str: openArray[char]; options: set[ParseFloatOptions];
+      sep = ','; decimalDot = '.'): float =
+    ## Convenience func for `parseFloat` which allows for thousand separators,
+    ## this is designed to parse floats as found in the wild formatted for humans.
+    ##
+    ## Fine grained flexibility and strictness is up to the user,
+    ## you can set the `options` using `ParseFloatOptions` enum.
+    ##
+    ## `parseFloatThousandSep` "prepares" `str` and then calls `parseFloat`,
+    ## consequently `parseFloatThousandSep` by design is slower than `parseFloat`.
+    ##
+    ## The following assumptions and requirements must be met:
+    ## - `sep` must not be `'-'`.
+    ## - `decimalDot` must not be `'-'` nor `' '`.
+    ## - `sep` and `decimalDot` must be different.
+    ## - `str` must be stripped of trailing and leading whitespaces.
+    ## - `pfTrailingDot` and `pfLeadingDot` must not be used together.
+    ##
+    ## See also:
+    ## * `parseFloat <strutils.html#parseFloat,string>`_
+    runnableExamples:
+      doAssert parseFloatThousandSep("10,000.000", {}) == 10000.0
```

Comment:
1st 2 lines redundant with 3rd line from a user perspective (but fine to include in ttstrmisc)